spyderlib/plugins/shortcuts.py

```diff
@@ -7,179 +7,505 @@
+        self.shortcuts = shortcuts
+
+        # Widgets
+        self.label_info = QLabel(_("Press the new shortcut and select 'Ok': \n"
+                                   "(Press 'Tab' once to switch focus between "
+                                   "shortcut entry and buttons)"))
+        self.label_current_sequence = QLabel(_("Current shortcut:"))
+        self.text_current_sequence = QLabel(sequence)
+        self.label_new_sequence = QLabel(_("New shortcut:"))
+        self.text_new_sequence = CustomLineEdit(self)
+        self.text_new_sequence.setPlaceholderText(sequence)
+        self.helper_button = HelperToolButton()
+
+        bbox = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        self.button_ok = bbox.button(QDialogButtonBox.Ok)
+        self.button_cancel = bbox.button(QDialogButtonBox.Cancel)
+
+        # Setup widgets
+        self.setWindowTitle(_('Shortcut: {0}').format(name))
+        self.button_ok.setFocusPolicy(Qt.NoFocus)
+        self.button_ok.setEnabled(False)
+        self.button_cancel.setFocusPolicy(Qt.NoFocus)
+        self.helper_button.setToolTip('')
+        self.helper_button.setFocusPolicy(Qt.NoFocus)
+        style = """
+            QToolButton {
+              margin:1px;
+              border: 0px solid grey;
+              padding:0px;
+              border-radius: 0px;
+            }"""
+        self.helper_button.setStyleSheet(style)
+
+        # Layout
+        spacing = 24
+        layout_sequence = QGridLayout()
+        layout_sequence.addWidget(self.label_info, 0, 0, 1, 3)
+        layout_sequence.addItem(QSpacerItem(spacing, spacing), 1, 0, 1, 2)
+        layout_sequence.addWidget(self.label_current_sequence, 2, 0)
+        layout_sequence.addWidget(self.text_current_sequence, 2, 2)
+        layout_sequence.addWidget(self.label_new_sequence, 3, 0)
+        layout_sequence.addWidget(self.helper_button, 3, 1)
+        layout_sequence.addWidget(self.text_new_sequence, 3, 2)
+
+        layout = QVBoxLayout()
+        layout.addLayout(layout_sequence)
+        layout.addSpacing(spacing)
+        layout.addWidget(bbox)
+        self.setLayout(layout)
+
+        # Signals
+        bbox.accepted.connect(self.accept)
+        bbox.rejected.connect(self.reject)
+
+    def keyPressEvent(self, e):
+        """Qt override"""
+        key = e.key()
+
+        self.npressed += 1
+        self.key_non_modifiers.append(key)
+        self.key_modifiers.add(key)
+        self.key_text.append(e.text())
+
+        debug_print('key %s, npressed: %s' % (key, self.npressed))
+
+        if key == Qt.Key_unknown:
+            return
+
+        # The user clicked just and only the special keys
+        # Ctrl, Shift, Alt, Meta.
+        if (key == Qt.Key_Control or
+                key == Qt.Key_Shift or
+                key == Qt.Key_Alt or
+                key == Qt.Key_Meta):
+            return
+
+        # Check if valid keys
+        if key not in VALID_KEYS:
+            return

```

Comment:
This is not working for *dead* keys (as @SylvainCorlay discovered). I mean, dead keys are the ones that need a further key stroke to make them appear, like `<tilde> e` to get `é`

So right now you can enter shortcuts like:

    Ctrl + <tilde> + e

and it'll be accepted (although they do nothing in practice)